client/shared/services/auth_service.coffee

```diff
@@ -11,23 +12,40 @@ module.exports = new class AuthService
       @applyEmailStatus(user, _.first(data.users))
 
   applyEmailStatus: (user, data = {}) ->
-    keys = ['name', 'email', 'avatar_kind', 'avatar_initials', 'email_hash', 'avatar_url', 'has_password', 'email_status']
+    keys = ['name', 'email', 'avatar_kind', 'avatar_initials', 'email_hash', 'avatar_url', 'has_password', 'email_status', 'email_verified', 'legal_accepted_at']
     user.update _.pick(_.mapKeys(_.pick(data, keys), (v,k) -> _.camelCase(k)), _.identity)
     user.update(hasToken: data.has_token)
     user
 
+  signUpOrIn: (user) ->
+    if user.emailStatus == 'unused'
+      @signUp(user)
+    else
+      @signIn(user)
+
   signIn: (user = {}) ->
-    Records.sessions.build(name: user.name, email: user.email, password: user.password).save()
+    Records.sessions.build
+      name: user.name
+      email: user.email
+      password: user.password
+      legalAccepted: user.legalAccepted
+    .save().then ->
+      hardReload()
```

Comment:
I'm hesitant to introduce the dep on `window` in AuthService, because native won't have a window, which means we have to rewrite AuthService... so I'd rather keep the window requires to a minimum if possible.